test/e2e/basic/csvs.go

```diff
@@ -0,0 +1,228 @@
+				err = cli.Create(context.TODO(), &element)
+				Expect(err).To(Succeed())
+				created++
+			}
+			Expect(created).To(Equal(existingCSVs))
+		})
+
+		When("there is no CSVs in the cluster", func() {
+			It("Should list CSVs without timeout", func() {
+				options := []client.ListOption{}
+				ret, err := paocontroller.ListPerformanceOperatorCSVs(cli, options, paginationLimit)
+				Expect(err).To(Succeed())
+				Expect(ret).To(HaveLen(0))
+
+			})
+		})
+
+		When("there are few CSVs in the cluster", func() {
+			const csvs_file = "../testing_manifests/csv-dummy.yaml"
+			const numberCSVsToCreate = 100
+			var csvs_created []types.NamespacedName
+			BeforeEach(func() {
+				var err error
+
+				By("reading file")
+				b, err := os.ReadFile(csvs_file)
+				Expect(err).To(Succeed())
+
+				By("unmarshaling file")
+				csvsOrig := olmv1alpha1.ClusterServiceVersion{}
+				err = yaml.Unmarshal(b, &csvsOrig)
+				Expect(err).To(Succeed())
+
+				By(fmt.Sprintf("Create some(%d) CSVs", numberCSVsToCreate))
+				for idx := 0; idx < numberCSVsToCreate; idx++ {
+					csvs := csvsOrig.DeepCopy()
+					csvs.Name = fmt.Sprintf("csvs-test-%d", idx)
+					csvs.Namespace = ntoconfig.WatchNamespace()
+
+					err = cli.Create(context.TODO(), csvs)
+					Expect(err).To(Succeed())
+					key := types.NamespacedName{
+						Name:      csvs.Name,
+						Namespace: csvs.Namespace,
+					}
+					csvs_created = append(csvs_created, key)
+				}
+				Expect(len(csvs_created)).To(Equal(numberCSVsToCreate))
+			})
+
+			AfterEach(func() {
+				var err error
+
+				By(fmt.Sprintf("Delete all(%d) previously created CSVs", len(csvs_created)))
+				for _, key := range csvs_created {
+					obj := olmv1alpha1.ClusterServiceVersion{}
+					err = cli.Get(context.TODO(), key, &obj)
+					Expect(err).To(Succeed())
+
+					err = cli.Delete(context.TODO(), &obj)
+					Expect(err).To(Succeed())
+				}
+				csvs_created = nil
+			})
+
+			When("pagination value is set to ONE", func() {
+				var oldPaginationLimit uint64
+				BeforeEach(func() {
+					oldPaginationLimit = paginationLimit
+					paginationLimit = 1
+				})
+				AfterEach(func() {
+					paginationLimit = oldPaginationLimit
+				})
+				It("Should list CSVs without timeout event with low pagination", func() {
+					options := []client.ListOption{}
+					ret, err := paocontroller.ListPerformanceOperatorCSVs(cli, options, paginationLimit)
+					Expect(err).To(Succeed())
+					Expect(ret).To(HaveLen(0))
+				})
+			})
+
+			When("pagination value is set to a much greater value than the CVS number", func() {
+				var oldPaginationLimit uint64
+				BeforeEach(func() {
+					oldPaginationLimit = paginationLimit
+					times := 100
+					paginationLimit = uint64(times * len(csvs_created))
+					By(fmt.Sprintf("Setting paginationLimit to (%d), that is %d times number of csvs created (%d)", paginationLimit, times, len(csvs_created)))
+				})
+				AfterEach(func() {
+					paginationLimit = oldPaginationLimit
+				})
```

Comment:
same comment about intentional shadowing

Reply:
I find more readable the "backup&restore" operation cause I can see it is a deliberate operation while shadowing could be a typo error ... but in this case shadowing looks better.